### PR TITLE
Admin page

### PR DESCRIPTION
**Pull request by EliasPinno**
Created an admin page with admin information. In the future, success rate by question category could be useful to include, when API support is enabled.

### CODE REVIEW

File: src/app/_models/category_stats.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {Category} from '@app/_models/category';
+
+export interface CategoryStats{
```
Comment (keyvankhademi) [inline src/app/_models/category_stats.ts:3]: This model only keeps the categories in a nested format. So it should have a better name (e.g NestedCategories) and should be moved to the category model file.

File: src/app/_models/question_counts.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface QuestionCount{
+    name: string;
+    count: number;
+    count_per_difficulty: DifficultyCount[];
+}
+interface DifficultyCount{
```
Comment (keyvankhademi) [inline src/app/_models/question_counts.ts:6]: There is no need to have DifficultyCount as a separate interface please have it inline in the QuestionCount interface.
```
export interface QuestionCount{
    name: string;
    count: number;
    count_per_difficulty: {count: number; difficulty: string}[];
}
```

File: src/app/_services/api/admin/category-stats.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
```
Comment (keyvankhademi) [inline src/app/_services/api/admin/category-stats.service.spec.ts:1]: This file needs to be formatted.

File: src/app/_services/api/admin/question-count.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { TestBed } from '@angular/core/testing';
```
Comment (keyvankhademi) [inline src/app/_services/api/admin/question-count.service.spec.ts:1]: This file needs to be formatted.

File: src/app/_services/api/admin/question-count.service.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import {Injectable} from '@angular/core';
```
Comment (keyvankhademi) [inline src/app/_services/api/admin/question-count.service.ts:1]: There is no need to have 2 separate files for one api endpoint. You should have 1 file (e.g admin.service.ts) and have both functions in that file.

File: src/app/_services/api/user-stats.service.ts [hunk 6]
```diff
@@ -24,6 +24,12 @@ export class UserStatsService {
             .pipe(catchError(this.handleError<UserStats>(`getUserStat`)));
     }
 
+    getAllUserStat(): Observable<UserStats[]> {
```
Comment (keyvankhademi) [inline src/app/_services/api/user-stats.service.ts:27]: Are you using this? Should be removed if you are not.

File: src/app/components/admin-components/admin/admin.component.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import {UserStatsService} from '@app/_services/api/user-stats.service';
+import {UserStats} from '@app/_models';
+
+@Component({
+    selector: 'app-admin',
+    templateUrl: './admin.component.html',
+    styleUrls: ['./admin.component.scss']
+})
+export class AdminComponent implements OnInit {
+    questionAPIData: QuestionCount[];
```
Comment (keyvankhademi) [inline src/app/components/admin-components/admin/admin.component.ts:16]: questionAPIData is a bad name. questionData or questionCount would be better.

Comment (keyvankhademi) [reply src/app/components/admin-components/admin/admin.component.ts:16]: Same for other members.

File: src/app/components/admin-components/admin/admin.component.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    categoryAPIData: CategoryStats[];
+    userStatAPIData: UserStats[];
+
+    constructor(private categoryStatsService: CategoryStatsService,
+                private questionCountService: QuestionCountService,
+                private userStatsService: UserStatsService) {
+    }
+
+    ngOnInit(): void {
+        this.questionCountService
```
Comment (keyvankhademi) [inline src/app/components/admin-components/admin/admin.component.ts:26]: This expanded form reduces the readability of the code. Please keep it in one line.

File: src/app/components/admin-components/category-stats/category-stats.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (keyvankhademi) [inline src/app/components/admin-components/category-stats/category-stats.component.spec.ts:1]: This file needs to be formatted.

File: src/app/_services/api/user-stats.service.ts (hunk 6 above)
Comment (EliasPinno) [reply src/app/_services/api/user-stats.service.ts]: Removed.

File: src/app/components/admin-components/admin/admin.component.ts (hunk 7 above)
Comment (EliasPinno) [reply src/app/components/admin-components/admin/admin.component.ts]: Renamed in admin component

File: src/app/_services/api/admin/question-count.service.ts (hunk 5 above)
Comment (EliasPinno) [reply src/app/_services/api/admin/question-count.service.ts]: I've combined both services into admin.service.ts

File: src/app/_models/category_stats.ts (hunk 1 above)
Comment (EliasPinno) [reply src/app/_models/category_stats.ts]: Model has been moved to Category, and renamed to NestedCategories. Rest of code has been refactored to reflect this.

File: src/app/_models/question_counts.ts (hunk 2 above)
Comment (EliasPinno) [reply src/app/_models/question_counts.ts]: Did exactly this.

File: src/app/_services/api/admin/category-stats.service.spec.ts (hunk 3 above)
Comment (EliasPinno) [reply src/app/_services/api/admin/category-stats.service.spec.ts]: This file was deleted in reformatting, so no need to.

File: src/app/_services/api/admin/question-count.service.spec.ts (hunk 4 above)
Comment (EliasPinno) [reply src/app/_services/api/admin/question-count.service.spec.ts]: This file was deleted in refactoring, so no need to.

File: src/app/components/admin-components/category-stats/category-stats.component.spec.ts (hunk 9 above)
Comment (EliasPinno) [reply src/app/components/admin-components/category-stats/category-stats.component.spec.ts]: Autoformatted all remaining files.

File: src/app/components/admin-components/admin/admin.component.ts (hunk 8 above)
Comment (EliasPinno) [reply src/app/components/admin-components/admin/admin.component.ts]: Sorry, I am actually not sure what you are talking about here. Do you mean the empty constructors, or do you mean how I subscribe the variables? Need more context so I can fix this. The multi line empty constructors is how webstorm auto formats it, and the multi line subscribe is how I saw it done in many other components.

**Pull request closed without merging**